kernel/term_typing.ml

```diff
@@ -90,15 +90,16 @@ let infer_declaration env (dcl : constant_entry) =
                         prim_entry_content = op_t;
                       }) ->
       let env = push_context_set ~strict:true uctxt env in
+      let u = Univ.Instance.empty (* FIXME *) in
```

Comment:
Working with the universes and explicit type in the polymorphic case is possible but I don't see a clean way to do it.
Are we very attached to the syntax with the type explicitly given? If not IMO we should just remove it and simplify the code.
cc @maximedenes

Reply:
We figured it out eventually.